curriculum/challenges/english/08-coding-interview-prep/rosetta-code/last-letter-first-letter.english.md

```diff
@@ -0,0 +1,129 @@
+---
+id: 5a23c84252665b21eecc7edd
+title: Last letter-first letter
+challengeType: 5
+---
+
+## Description
+<section id='description'>
+A certain children's game involves starting with a word in a particular category. Each participant in turn says a word, but that word must begin with the final letter of the previous word. Once a word has been given, it cannot be repeated. If an opponent cannot give a word in the category, they fall out of the game.
+For example, with "animals" as the category,
+<pre>
+Child 1: dog
+Child 2: goldfish
+Child 1: hippopotamus
+Child 2: snake
+...
+</pre>
+</section>
+
+## Instructions
+<section id='instructions'>
+Write a function that takes an input array of words. The function should return an array of words where the first letter of each word is the same as the last letter of the previous word. The first word in the return array can be anything. Use only the words in the input array, and once a word is used it cannot be repeated. The words in the return array should be selected and sequenced so that that the length of the array is maximized.
+</section>
+
+## Tests
+<section id='tests'>
+
+``` yml
+tests:
+  - text: <code>findLongestChain</code> should be a function.
+    testString: assert(typeof findLongestChain == 'function', '<code>findLongestChain</code> should be a function.');
+  - text: <code>findLongestChain(["certain", "each", "game", "involves", "starting", "with", "word"])</code> should return an array.
+    testString: assert(Array.isArray(findLongestChain(["certain", "each", "game", "involves", "starting", "with", "word"])), '<code>findLongestChain(["certain", "each", "game", "involves", "starting", "with", "word"])</code> should return an array.');
```

Comment:
This is optional and more of a note for future PRs, but we no longer need the last argument for the `testString`s. This one can be shortened to:

```
testString: assert(Array.isArray(findLongestChain(["certain", "each", "game", "involves", "starting", "with", "word"])));
```

But again, this change is definitely not required for this PR.